<launch>

  <arg name="bag_path" default="/home/henryzh47/logging/2019-05-24-toured/objdet_2019-05-24-13-02-04_5.bag"/>

  <node pkg="image_proc" name="image_proc" type="image_proc" ns="thermal"/>

  <remap from="/thermal" to="/thermal/image_raw"/>

  <node pkg="rosbag" type="play" name="bag_player" args="$(arg bag_path) --clock"/>

  <node pkg="flir_ros" name="add_caminfo_to_bag" type="add_caminfo_to_bag.py" output="screen">
    <param name="intrinsic_url" value="$(find flir_ros)/data/camera_info/thermal_caminfo_flir_320.yaml"/>
    <param name="thermal_topic" value="/thermal"/>
    <param name="camera_info_topic" value="/thermal/camera_info"/>
    <param name="thermal_frame_id" value="thermal/optical_frame"/>
  </node>

   <!-- ############ tf ############### -->
  <node pkg="tf" type="static_transform_publisher" name="vlp_to_thermal_camera_link"
        args="0.0868 0.0161 -0.1621 0.0126  0.0721 0.0396 /velodyne /thermal/camera_link 1000"/>

  <node pkg="tf" type="static_transform_publisher" name="thermal_camera_link_to_optical"
        args="0 0 0 -1.57079632679 0 -1.57079632679 /thermal/camera_link /thermal/optical_frame 1000"/>
    
  <include file="$(find flir_ros)/launch/disable_rect_topics.launch"/>

</launch>
